src/components/implementation/no_interface/llbooter/llbooter.c

```diff
@@ -146,11 +191,47 @@ comps_init(void)
 			if (crt_comp_create(comp, name, id, elf_hdr, info)) {
 				printc("Error constructing the resource tables and image of component %s.\n", comp->name);
 				BUG();
-			}	
+			}
 		}
 		assert(comp->refcnt != 0);
 	}
 
+	/* allocate, initialize, and add components to namespaces */
+	struct crt_ns_asid *ns_asid = ss_ns_asid_alloc();
+	assert(ns_asid);
+	ss_ns_asid_activate(ns_asid);
+
+	if(crt_ns_asids_init(ns_asid) != 0) BUG();
+
+	struct crt_ns_vas *ns_vas1 = ss_ns_vas_alloc();
+	assert(ns_vas1);
+	ss_ns_vas_activate(ns_vas1);
+
+	if(crt_ns_vas_init(ns_vas1, ns_asid) != 0) BUG();
+
+	/* FIXME: THIS IS HARD CODED FOR THE ping_pong_shared_vas.toml COMPOSITION SCRIPT */
```

Comment:
I think I see an issue with creating the namespaces after component creation. crt_comp_create is responsible for aliasing memory into a component; when using MPK, this requires knowledge of the MPK key (since we cant modify the pgtbl flags after aliasing the memory). We probably have to create the namespaces before this happens.